server/middleware/CheckRoleMiddleware.js

```diff
@@ -0,0 +1,11 @@
+function checkRole(roleArray) {
+  return function(req, res, next) {
+    // Check if the user exists and if their role is included in the provided roleArray
+    if (req.user && roleArray.includes(req.user.role)) {
```

Comment:
This will not work, since you are no longer appending a `user` object.

Refactor.

Reply:
done

Reply:
is this tested?